### PR TITLE
Define compiler env vars for CMake in dev containers.

### LINKED ISSUE #575: [BUG]: Using CMake presets in a devcontainer does not properly configure compilers.
### Is this a duplicate?

- [X] I confirmed there appear to be no [duplicate issues](https://github.com/NVIDIA/cccl/issues) for this bug and that I agree to the [Code of Conduct](CODE_OF_CONDUCT.md)

### Type of Bug

Something else

### Component

Not sure

### Describe the bug

While the CI scripts are working as expected, configuring a preset directly in vscode from a dev container does not properly setup the dev container compilers.

### How to Reproduce

Configure a preset in vscode from a dev container. The nvcc invocations will not set --ccbin.

### Expected behavior

The compilers should be configured in the dev container environment using the variables that CMake expects, namely `CXX` and `CUDAHOSTCXX`.

### Reproduction link

_No response_

### Operating System

_No response_

### nvidia-smi output

_No response_

### NVCC version

_No response_

### PR DESCRIPTION
**Pull request by allisonvacanti**
## Description

closes #575

Adds compilers to the devcontainer environments.

### CODE REVIEW

**Review by miscco — APPROVED** [review]

File: .devcontainer/cuda12.2-gcc8/devcontainer.json [hunk 1]
```diff
@@ -18,7 +18,9 @@
     "CCCL_CUDA_VERSION": "12.2",
     "CCCL_HOST_COMPILER": "gcc",
     "CCCL_HOST_COMPILER_VERSION": "8",
-    "CCCL_BUILD_INFIX": "cuda12.2-gcc8"
+    "CCCL_BUILD_INFIX": "cuda12.2-gcc8",
+    "CXX": "gcc",
+    "CUDAHOSTCXX": "gcc"
```
[the remark this answers is not in the document]
Comment (allisonvacanti) [reply .devcontainer/cuda12.2-gcc8/devcontainer.json:23]: That would work when using devcontainers, but that variable will not be defined in all cases (e.g., configuring outside of a devcontainer). CMake directly checks the CXX and CUDAHOSTCXX, so setting those will always work. This is also how the CI scripts configure the compilers, so it's nice for consistency.

**Review by senior-zero — APPROVED** [review]
Build through the CMake extension works now, thanks! One minor note is that it's using make as generator. Any chance we could set `CMAKE_GENERATOR` to Ninja in devcontainers?

**Comment by allisonvacanti** [comment]
> Build through the CMake extension works now, thanks! One minor note is that it's using make as generator. Any chance we could set CMAKE_GENERATOR to Ninja in devcontainers?

Strange...the presets are hardcoded to use ninja currently. I'll look into this.

**Comment by senior-zero** [comment]
My bad, Ninja is used indeed!

**Pull request merged**